src/mc-fle2-payload-iev-v2.c

```diff
@@ -429,9 +483,54 @@ bool mc_FLE2IndexedRangeEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2_t
     const uint64_t SEV_len = SEV_and_edges_len - edges_len;
     CHECK_AND_RETURN(mc_reader_read_buffer(&reader, &iev->ServerEncryptedValue, SEV_len, status));
 
-    // Ignore Metadata block.
-    BSON_ASSERT(mc_reader_get_remaining_length(&reader) == edges_len);
+    iev->metadata = (_mongocrypt_buffer_t *)malloc(iev->edge_count * sizeof(_mongocrypt_buffer_t));
```

Comment:
I wasn't sure whether to allocate memory like this or to initialize the array to its max size.
``` C
_mongocrypt_buffer_t edges[255]; // max uint8_t
```

Reply:
Initializing to a large, but still arbitrary size is definitely not the right move.  I think having a pointer with its own array allocation is fine here. There is a neat trick you can do with C structs where an unsized array dangles off the end, but it's not one that'll work well here since we don't know the size till parsing starts and we don't want the caller to have to think about allocation sizing.

TL;DR - I'm okay with this plan,

But... per my comment on #899, you should be using `bson_malloc0` here.